### PR TITLE
New processiong for partners

### PR DESCRIPTION
**Pull request by UncleAndy**

### CODE REVIEW

File: cache/cache.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+var redis_client *redis.Client
+
+func Init() {
+	redis_addr := os.Getenv("REDIS_ADDR")
+	redis_pass := os.Getenv("REDIS_PASS")
+	redis_db   := os.Getenv("REDIS_DB")
+
+	if redis_addr == "" {
+		redis_addr = "localhost:6379"
+	}
```
Comment (evgenTraytyak) [inline cache/cache.go]: put empty line after `if` statement

File: cache/cache.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	if redis_db == "" {
+		redis_db = "0"
+	}
```
Comment (evgenTraytyak) [inline cache/cache.go]: put empty line after `if` statement

File: cache/cache.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	redis_db_num, err := strconv.Atoi(redis_db)
```
Comment (evgenTraytyak) [inline cache/cache.go]: put empty line after variable assignment

File: cache/partners_tours.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	io.WriteString(h, strconv.Itoa(tour.CountryId))
+	io.WriteString(h, strconv.Itoa(tour.TownId))
+	io.WriteString(h, strconv.Itoa(tour.Adults))
+	io.WriteString(h, tour.Checkin)
+	io.WriteString(h, strconv.Itoa(tour.Nights))
+	io.WriteString(h, strconv.Itoa(tour.Kids))
+	kid1age := -1
+	if tour.Kid1Age != nil {
+		kid1age = *(tour.Kid1Age)
+	}
```
Comment (evgenTraytyak) [inline cache/partners_tours.go]: put empty after and before `if` statement everywhere

File: data/tour.go [hunk 5]
```diff
@@ -1,21 +1,36 @@
...
+	RequestId		int `xml:"requestId,attr"`
+	OfferId			int64 `xml:"offerId,attr"`
+	FewEconomTicketsDpt	int `xml:"fewEconomTicketsDpt,attr"`
+	FewEconomTicketsRtn	int `xml:"fewEconomTicketsRtn,attr"`
+	FewPlacesInHotel	int `xml:"fewPlacesInHotel,attr"`
+	Flags			int64 `xml:"flags,attr"`
+	Description		string `xml:"description,attr"`
+	TourUrl			string `xml:"tourUrl,attr"`
+	RoomName		string `xml:"roomName,attr"`
+	ReceivingParty		string `xml:"receivingParty,attr"`
```
Comment (evgenTraytyak) [inline data/tour.go]: align third arguments to one column

File: stat/tours.go [hunk 6]
```diff
@@ -1,12 +1,25 @@
...
 	tours := &Tours{
+		total: 0,
 		Total:     make(chan uint64),
+		skipped: 0,
 		Skipped:   make(chan uint64),
+		kidsIssue: 0,
 		KidsIssue: make(chan uint64),
+		End: make(chan bool),
+		historyUsed: 0,
+		currentHour: "",
```
Comment (evgenTraytyak) [inline stat/tours.go]: align value assignments in one column or just use space after `:`

File: cache/partners_tours.go [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+}
+
+func SaveTourGroupsToDB(once_flag *sync.Once) {
+	count := redis_client.LLen("pt_tours_groups_keys").Val()
+	log.Info.Println("SaveTourGroupsToDB START (", count, ")...")
+
+	batch_size := savePdCommitBatchSize
+	transaction, trx_err := db.StartTransaction()
+	for row := redis_client.LPop("pt_tours_groups_keys");
+		row.Err() == nil && trx_err == nil;
```
Comment (UncleAndy) [inline cache/partners_tours.go]: Последовательная выборка значений из списка.

**Pull request merged**